elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -107,6 +123,64 @@ else if (queryObj.getQueryType().equals(QueryType.GRAPHQL_V1_0)) {
         }
     }
 
+    /**
+     * Checks if the response obtained from the query is in JSON format.
+     * @param test response.getBody() from the AsyncQuery response
+     * @return returns true/ false
+     */
+    protected boolean isValidJSON(String test) {
+        try {
+            new JSONObject(test);
+        } catch (JSONException ex) {
+            try {
+                new JSONArray(test);
+            } catch (JSONException ex1) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    protected Integer calculateRecordsJSON(String jsonStr) {
+        Integer rec;
+        try {
+            JSONObject j = new JSONObject(jsonStr);
+            try {
+                rec = j.getJSONArray("data").length();
+            } catch (JSONException e2) {
+                rec = null;
+            }
+        } catch (JSONException e) {
+            rec = null;
+        }
+        return rec;
+    }
+
+    protected Integer calculateRecordsGRAPHQL(String jsonStr) {
+        Integer rec;
+        try {
+            JSONObject j = new JSONObject(jsonStr);
+            try {
+                JSONObject j2 = j.getJSONObject("data");
+                try {
+                    JSONObject j3 = j2.getJSONObject("book");
```

Comment:
we should not have book here. It will be different based on different objects,

Reply:
Changed the logic to get the tableName from the AsyncQuery.query. 
Details in the latest commit